homeassistant/components/climate/zhong_hong.py

```diff
@@ -0,0 +1,166 @@
+"""
+Support for ZhongHong HVAC Controller.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.zhong_hong/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA, SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (ATTR_TEMPERATURE, CONF_HOST, CONF_PORT,
+                                 TEMP_CELSIUS)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.temperature import convert as convert_temperature
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GW_ADDR = 'gw_addr'
+
+REQUIREMENTS = ['zhong_hong_hvac==1.0.0']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST):
+    cv.string,
+    vol.Optional(CONF_PORT, default=9999):
+    vol.Coerce(int),
+    vol.Optional(CONF_GW_ADDR, default=1):
+    vol.Coerce(int),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZhongHong HVAC platform."""
+    from zhong_hong_hvac.hub import ZhongHongGateway
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    gw_addr = config.get(CONF_GW_ADDR)
+    hub = ZhongHongGateway(host, port, gw_addr)
+    devices = [
+        ZhongHongClimate(hub, addr_out, addr_in)
+        for (addr_out, addr_in) in hub.discovery_ac()
+    ]
+    add_devices(devices)
+    hub.start_listen()
+    hub.query_all_status()
+
+
+class ZhongHongClimate(ClimateDevice):
+    """Representation of a ZhongHong controller support HVAC."""
+
+    def __init__(self, hub, addr_out, addr_in):
+        """Setup platform."""
```

Comment:
This should be in imperative mood. Perhaps something like: "Set up the ZhongHong climate devices."